docs/src/pages/en/jupyter-widgets-the-good-parts.mdx

```diff
@@ -291,6 +291,128 @@ widget
+
+You might be curious how `traitlets` come into play with data types. Although
+`ipywidgets` are deeply tied to `traitlets`, it's just a library to help with
+validation and custom serialization if necessary. Ultimately, all data sent to
+the frontend must match a Python data type in the table above. 
+
+```python
+class Widget(anywidget.AnyWidget):
+    _esm = """
+    function render({ model, el }) {
+      console.log(model.get("my_str"));    // "Hello, World!"
+      console.log(model.get("my_float"));  // 3.14
+      console.log(model.get("my_int"));    // 42
+      console.log(model.get("my_bool"));   // true
+      console.log(model.get("my_dict"));   // { foo: "bar", bar: 42 }
+      console.log(model.get("my_list"));   // [ "foo", "bar", 42 ]
+      console.log(model.get("my_set"));    // [ "foo", "bar", 42 ]
+      console.log(model.get("my_bytes"));  // DataView(13)
+    }
+    export default { render };
+    """
+    my_str = traitlets.Unicode("Hello, World!").tag(sync=True)
+    my_float = traitlets.Float(3.14).tag(sync=True)
+    my_int = traitlets.Int(42).tag(sync=True)
+    my_bool = traitlets.Bool(True).tag(sync=True)
+    my_dict = traitlets.Dict({"foo": "bar", "bar": 42}).tag(sync=True)
+    my_list = traitlets.List(["foo", "bar", 42]).tag(sync=True)
+    my_set = traitlets.Set({"foo", "bar", 42}).tag(sync=True)
+    my_bytes = traitlets.Bytes(b"Hello, World!").tag(sync=True)
+```
+
+The specific traitlets above just provide validation on the Python side.
+Alternatively, you can just use `traitlets.Any` to avoid validation, and
+the data will sill be serialized to the front end according to the table above.
+
+```python
+class Widget(anywidget.AnyWidget):
+    _esm = """
+    function render({ model, el }) {
+      model.on("change:whatever", () => {
+        console.log(model.get("whatever"));
+      })
+    }
+    export default { render };
+    """
+    whatever = traitlets.Any().tag(sync=True)
+
+w = Widget()
+w
+```
+
+```py
+w.whatever = "Hello, World!"            # "Hello, World!"
+w.whatever = 3.14                       # 3.14
+w.whatever = 42                         # 42
+w.whatever = True                       # true
+w.whatever = {"foo": "bar", "bar": 42}  # { foo: "bar", bar: 42 }
+w.whatever = ["foo", "bar", 42]         # [ "foo", "bar", 42 ]
+w.whatever = {"foo", "bar", 42}         # [ "foo", "bar", 42 ]
+w.whatever = b"Hello, World!"           #  DataView(13)
+```
+
+## Custom Serialization
+
+A custom serializer for a trait can be defined by implementing `to_json` hook
+on the trait metadata, which must return one of the types listed in the [Data
+Types](#data-types) table.
+
+For example, let's serialize a `pathlib.Path` to it's file contents:
+
+```python
+import pathlib
+
+def path_to_json(path: pathlib.Path, widget: anywidget.AnyWidget):
```

Comment:
It seems like the widget argument is not used in the function body - perhaps add a comment if there is a reason behind the function signature

Reply:
Good idea. I'm actually not sure why it's there (ipywidgets thing), I guess if you needed some other widget state/method for the serialization. But with Python you need to implement the full signature otherwise the caller will get an error.